src/SOS/Strike/sigparser.cpp

```diff
@@ -0,0 +1,166 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#ifndef FEATURE_PAL
+#include <tchar.h>
+#endif
+#include "strike.h"
+#include "util.h"
+
+#ifndef IfFailRet
+#define IfFailRet(EXPR) do { HRESULT Status = (EXPR); if(FAILED(Status)) { return (Status); } } while (0)
+#endif
+
+HRESULT SigParser::SkipExactlyOne()
+{
+    CorElementType typ;
+    HRESULT hr = GetElemType(&typ);
+    
+    IfFailRet(hr);
+        
+    if (!CorIsPrimitiveType(typ))
+    {
+        switch ((DWORD)typ)
+        {
+            default:
+                // _ASSERT(!"Illegal or unimplement type in COM+ sig.");
+                return META_E_BAD_SIGNATURE;
+                break;
+            case ELEMENT_TYPE_VAR:
+            case ELEMENT_TYPE_MVAR:
+                IfFailRet(GetData(NULL));      // Skip variable number
+                break;
+/*            case ELEMENT_TYPE_VAR_ZAPSIG:
```

Comment:
I noticed a lot of these cases were commented out unlike the sigparser.cpp code in coreclr. SOS will never see them?

Reply:
SOS should never see them, and in particular the commented out cases are internal implementation details that may change based on the version of the CLR. (They haven't for years, but its always possible.)

Reply:
In particular, they will never show up in typespecs, or in method signatures which is all that is parsed here.